zcash_test_vectors/transaction.py

```diff
@@ -571,7 +556,19 @@ def __bytes__(self):
         ret += super().to_bytes(self.version_bytes(), self.nVersionGroupId, self.nConsensusBranchId)
 
         # Orchard Transaction Fields
+        ret += write_compact_size(len(self.vActionsOrchard))
         if len(self.vActionsOrchard) > 0:
+            # Not explicitly gated in the protocol spec, but if the gate
+            # were inactive then these loops would be empty by definition.
+            for desc in self.vActionsOrchard:
+                ret += bytes(desc) # Excludes spendAuthSig
+            ret += struct.pack('B', self.flagsOrchard)
+            ret += struct.pack('<Q', self.valueBalanceOrchard)
+            ret += bytes(self.anchorOrchard)
+            ret += write_compact_size(len(self.proofsOrchard))
+            ret += self.proofsOrchard
+            for desc in self.vActionsOrchard:
+                ret += bytes(desc.spendAuthSig)
```

Comment:
why is this change needed? why not keep it in `to_bytes()`?

Reply:
Because now V6 transactions work per Action Group. TransactionBase needs to only have the things that are common between V5 and V6, and when V6 works per Action Group it will not generalize to greater than 1 Action Group without moving it out of TransactionBase.

Even when there is only one Action Group the nAGExpiryHeight field comes in between proofsOrchard and the spendAuthSig values.